tgui/packages/tgui/interfaces/common/temperatureUtils.js

```diff
@@ -0,0 +1,37 @@
+import { Color } from 'common/color';
+
+export const TemperatureColors = {
+  cold: new Color(66, 194, 255),
+  neutral: new Color(170, 170, 170),
+  hot: new Color(255, 120, 0),
+  veryhot: new Color(255, 0, 0),
+};
+
+const neutralTemperature = 293.15;
+const deviation = 200;
+const highTemperature = neutralTemperature + deviation;
+
+export const getTemperatureColor = (temperature) => {
+  const { cold, neutral, hot, veryhot } = TemperatureColors;
+
+  if (temperature < highTemperature) {
+    return Color.lookup((temperature - neutralTemperature) / (deviation * 2) + 0.5, [cold, neutral, hot]);
+  }
+  return Color.lookup((temperature - highTemperature) / (1000 - highTemperature), [hot, veryhot]);
+};
+
+export const getTemperatureIcon = (temperature) => {
+  switch (Math.round(temperature/200)) {
+    case (0): return "thermometer-empty";
+    case (1): return "thermometer-quarter";
+    case (2): return "thermometer-half";
+    case (3): return "thermometer-three-quarters";
+    default: return "thermometer-full";
+  }
+};
+
+export const getTemperatureChange = (temperature, targetTemperature) => {
```

Comment:
Function name here doesn't seem to correspond to behavior? I mean, it *does*, but it's not what I would expect a function of that name with those arguments to do.

`getTemperatureChangeName` or something to indicate that this is the label that goes with that change?